tempesta_fw/classifier/frang.c

```diff
@@ -482,18 +484,66 @@ frang_http_host_check(const TfwHttpReq *req, FrangAcc *ra)
 	 * have a good regex library.
 	 * For now just linearize the Host header field TfwStr{} string.
 	 */
-	len = field.len + 1;
-	if ((buf = tfw_pool_alloc(req->pool, len)) == NULL)
-		return TFW_BLOCK;
-	tfw_str_to_cstr(&field, buf, len);
-	if (!tfw_addr_pton(buf, &addr)) {
+
+	urlhostlen = req->host.len;
+	hdrhostlen = field.len;
+
+	if (urlhostlen) {
+		if ((urlhost = tfw_pool_alloc(req->pool, urlhostlen + 1)) == NULL) {
```

Comment:
Do we need to linearize `req->host`? Why can't we use `tfw_str_eq_cstr()` for comparison with `field` (`hdrhost`)?